### ISSUE TITLE
Failing WebGL renderer tests

### ISSUE THREAD

**Issue by ahocevar** [openlayers/openlayers#15735]
Several WebGL renderer tests are failing for me locally:
```
Chrome Headless 123.0.0.0 (Mac OS 10.15.7) ol.webgl.styleparser parseLiteralStyle icon style contains main properties and expressions, icon specified as data url sets up builder accordingly FAILED
	Error: expected [ 'vec2 u_texture980902294_size',
	  'sampler2D u_texture980902294' ] to sort of equal [ 'sampler2D u_texture980902294' ]
	    at Assertion.assert (/Users/ahocevar/projects/openlayers/node_modules/expect.js/index.js:96:13)
	    at Assertion.eql (/Users/ahocevar/projects/openlayers/node_modules/expect.js/index.js:230:10)
	    at Context.<anonymous> (/var/folders/8j/lpm7ppqs7zl9w1lmwgw4kxj00000gn/T/_karma_webpack_91171/webpack:/ol/test/browser/spec/ol/webgl/styleparser.test.js:330:1)
.......................
Chrome Headless 123.0.0.0 (Mac OS 10.15.7) ol.webgl.styleparser parseLiteralStyle stroke style stroke pattern, tint, spacing, offset and size sets the color expression FAILED
	Error: expected 'vec4(1.0, 0.0, 0.0, 1.0) * sampleStrokePattern(u_texture980902294, u_texture980902294_size, vec2(0., u_texture980902294_size.y) + vec2(5.0, 5.0) * vec2(0., -1.) + vec2(5.0, 10.0) * vec2(1., -1.), vec2(5.0, 5.0), (2.0 * 10.0), currentLengthPx, currentRadiusRatio, v_width)' to sort of equal 'vec4(1.0, 0.0, 0.0, 1.0) * sampleStrokePattern(u_texture980902294, vec2(1.0, 1.0), vec2(0., vec2(1.0, 1.0).y) + vec2(5.0, 5.0) * vec2(0., -1.) + vec2(5.0, 10.0) * vec2(1., -1.), vec2(5.0, 5.0), (2.0 * 10.0), currentLengthPx, currentRadiusRatio, v_width)'
	    at Assertion.assert (/Users/ahocevar/projects/openlayers/node_modules/expect.js/index.js:96:13)
	    at Assertion.eql (/Users/ahocevar/projects/openlayers/node_modules/expect.js/index.js:230:10)
	    at Context.<anonymous> (/var/folders/8j/lpm7ppqs7zl9w1lmwgw4kxj00000gn/T/_karma_webpack_91171/webpack:/ol/test/browser/spec/ol/webgl/styleparser.test.js:766:1)
....
Chrome Headless 123.0.0.0 (Mac OS 10.15.7) ol.webgl.styleparser parseLiteralStyle fill style fill pattern, tint, offset and size sets the color expression FAILED
	Error: expected 'vec4(1.0, 0.0, 0.0, 1.0) * sampleFillPattern(u_texture980902294, u_texture980902294_size, vec2(0., u_texture980902294_size.y) + vec2(5.0, 5.0) * vec2(0., -1.) + vec2(5.0, 10.0) * vec2(1., -1.), vec2(5.0, 5.0), pxOrigin, pxPos)' to sort of equal 'vec4(1.0, 0.0, 0.0, 1.0) * sampleFillPattern(u_texture980902294, vec2(1.0, 1.0), vec2(0., vec2(1.0, 1.0).y) + vec2(5.0, 5.0) * vec2(0., -1.) + vec2(5.0, 10.0) * vec2(1., -1.), vec2(5.0, 5.0), pxOrigin, pxPos)'
	    at Assertion.assert (/Users/ahocevar/projects/openlayers/node_modules/expect.js/index.js:96:13)
	    at Assertion.eql (/Users/ahocevar/projects/openlayers/node_modules/expect.js/index.js:230:10)
	    at Context.<anonymous> (/var/folders/8j/lpm7ppqs7zl9w1lmwgw4kxj00000gn/T/_karma_webpack_91171/webpack:/ol/test/browser/spec/ol/webgl/styleparser.test.js:873:1)
```
@jahow, can you please take a look at this? Let me know if you need more information. One thing that might be of inerest: my device has a pixel ratio of 2.

**Comment by jahow** [comment]
Thanks for the pointer, I'll have a look

**Comment by jahow** [comment]
Fixed by #15942
